jetty-io/src/main/java/org/eclipse/jetty/io/ConnectionStatistics.java

```diff
@@ -78,159 +65,312 @@ public void onOpened(Connection connection)
     {
         if (!isStarted())
             return;
-
-        _connections.increment();
+        _stats.incrementCount();
+        _statsMap.computeIfAbsent(connection.getClass().getName(), Stats::new).incrementCount();
     }
 
     @Override
     public void onClosed(Connection connection)
     {
         if (!isStarted())
             return;
+        onClosed(_stats, connection);
+        Stats stats = _statsMap.get(connection.getClass().getName());
```

Comment:
Ditto. Put this get call into a protected method so selection can be done on criteria other than class name.